geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/Runner.scala

```diff
@@ -148,32 +167,55 @@ trait Runner extends LazyLogging {
     FileUtils.writeStringToFile(file, out.toString(), StandardCharsets.UTF_8)
   }
 
-  protected def createCommands(jc: JCommander): Seq[Command]
+  /**
+   * Commands available to this runner. The default impl handles common commands
+   * and placeholders for script functions
+   *
+   * @return
+   */
+  protected def commands: Seq[Command] = {
+    Seq(
+      new ConvertCommand,
+      new ConfigureCommand,
+      new ClasspathCommand,
+      new EnvironmentCommand,
+      new GenerateAvroSchemaCommand,
+      new NailgunCommand,
+      new ScalaConsoleCommand,
+      new VersionCommand
+    )
+  }
+
   protected def resolveEnvironment(command: Command): Unit = {}
 
   class DefaultCommand(jc: JCommander) extends Command {
     override def execute(): Unit = Command.user.info(usage(jc))
     override val name: String = ""
     override val params: Any = null
   }
+
+  class MainExecutor(args: Array[String]) extends Executor {
+    override def execute(): Unit = parseCommand(args).execute()
+  }
+
+  class NailgunExecutor(context: NGContext) extends Executor {
+    override def execute(): Unit = {
+      val command = parseCommand(context.getArgs)
+      context.getNGServer match {
+        case ng: NailgunServer => ng.execute(command)
+        case ng => throw new IllegalStateException(s"Expected a NailgunServer but got: $ng")
+      }
+    }
+  }
 }
 
 object Runner {
 
-  private var added = false
-
-  /**
-   * Add jcommander converter factories. Note that the implementation is a static,
-   * shared, non-thread-safe list, so we have to synchronize ourselves.
-   *
-   * TODO revisit this if we upgrade jcommander versions
-   */
-  def addConverterFactories(jc: JCommander): Unit = synchronized {
```

Comment:
What happened to this?

Reply:
my TODO above came true! :D

Reply:
Yeah, I noticed the change in JCommander version at the end of the PR!